src/mlpack/methods/ann/make_alias.hpp

```diff
@@ -15,44 +15,128 @@
 #define MLPACK_METHODS_ANN_MAKE_ALIAS_HPP
 
 #include <mlpack/prereqs.hpp>
+#include <bandicoot>
 
 namespace mlpack {
 
+template<typename MatType>
+struct IsCootMatType;
+
+template<typename CubeType>
+struct IsCootCubeType;
+
+template<typename MatType>
+struct IsCootMatType
+{
+  constexpr static bool value = false;
+};
+
+template<typename CubeType>
+struct IsCootCubeType
+{
+  constexpr static bool value = false;
+};
+
+#ifdef MLPACK_HAS_COOT
+template<typename MatType>
+struct IsCootMatType
+{
+  constexpr static bool value = true;
+};
+
+template<typename CubeType>
+struct IsCootCubeType
+{
+  constexpr static bool value = true;
+};
+
+#endif
+
 /**
  * Reconstruct `m` as an alias around the memory `newMem`, with size `numRows` x
  * `numCols`.
  */
-template<typename MatType>
-void MakeAlias(MatType& m,
-               typename MatType::elem_type* newMem,
+template<typename InMatType,
+         typename OutMatType,
+         typename = typename std::enable_if<
+         !IsCootMatType<InMatType>::value>::type>
```

Comment:
```suggestion
             !IsCootMatType<InMatType>::value>::type>
```
(just a little style pedantry, in my view this helps it make clear that this is a wrapped line for a template parameter)